// @generated SignedSource<<bf277ba3668c3a312a4d45bbf6faece1>>
// DO NOT EDIT THIS FILE MANUALLY!
// This file is a mechanical copy of the version in the configerator repo. To
// modify it, edit the copy in the configerator repo instead and copy it over by
// running the following in your fbcode directory:
//
// configerator-thrift-updater glean/client/client_config.thrift

namespace cpp2 facebook.glean
namespace hs Glean
namespace py glean.client_config
namespace py3 glean
namespace php glean
namespace rust glean_client_config

include "glean/config/service.thrift"

enum UseShards {
  NO_SHARDS = 0,
  USE_SHARDS = 1,
  USE_SHARDS_AND_FALLBACK = 2
} (hs.prefix = "", hs.nounknown)

struct ClientConfig {
  1: service.Service serv = { "tier" : "glean.query.prod" },
    // The default tier to connect to

  2: UseShards use_shards = USE_SHARDS_AND_FALLBACK
    // Whether to use shards or not

  3: i32 min_db_age = 1800
    // The minimum age (in seconds) of a DB before we will consider
    // using it. The idea here is that if we wait a little while
    // there's a better chance that the DB will have propagated to all
    // the servers.

  4: i32 host_timeout_ms = 30000
    // The timeout to use when we connect to a Glean server using an
    // explicit host:port, otherwise Thrift defaults to 100ms.
}
